modules/mcc/include/opencv2/mcc/ccm.hpp

```diff
@@ -0,0 +1,493 @@
+    CMC_2TO1,
+    RGB,
+    RGBL
+};
+
+/** @brief Core class of ccm model.
+           produce a ColorCorrectionModel instance for inference.
+*/
+
+class CV_EXPORTS_W ColorCorrectionModel
+{
+public:
+        Mat CV_WRAP ccm;
+        /** @brief Color Correction Model
+            @param src detected colors of ColorChecker patches;\n
+                        the color type is RGB not BGR, and the color values are in [0, 1];
+                        type is cv::Mat;
+            @param constcolor the Built-in color card;\n
+                        Supported list:
+                                Macbeth(Macbeth ColorChecker) ;
+                                Vinyl(DKK ColorChecker) ;
+                                DigitalSG(DigitalSG ColorChecker with 140 squares);\n
+                        type: enum CONST_COLOR;\n
+        */
+    ColorCorrectionModel(Mat src, CONST_COLOR constcolor);
+     /** @brief Color Correction Model
+            @param src detected colors of ColorChecker patches;\n
+                        the color type is RGB not BGR, and the color values are in [0, 1];
+                        type is cv::Mat;
+            @param colors the reference color values,the color values are in [0, 1].\n
+                        type: cv::Mat
+            @param ref_cs the corresponding color space
+                        NOTICE: For the list of color spaces supported, see the notes above;\n
+                        If the color type is some RGB, the format is RGB not BGR;\n
+                        type:enum COLOR_SPACE;
+        */
+    ColorCorrectionModel(Mat src, Mat colors, COLOR_SPACE ref_cs);
+    /** @brief Color Correction Model
+            @param src detected colors of ColorChecker patches;\n
+                        the color type is RGB not BGR, and the color values are in [0, 1];
+                        type is cv::Mat;
+            @param colors the reference color values,the color values are in [0, 1].\n
+                        type: cv::Mat
+            @param ref_cs the corresponding color space
+                        NOTICE: For the list of color spaces supported, see the notes above;\n
+                        If the color type is some RGB, the format is RGB not BGR;\n
+                        type:enum COLOR_SPACE;
+            @param colored mask of colored color
+        */
+    ColorCorrectionModel(Mat src, Mat colors, COLOR_SPACE ref_cs, Mat colored);
+
+        /** @brief set ColorSpace
+            @param cs_ the absolute color space that detected colors convert to;\n
+            NOTICE: it should be some RGB color space;\n
+                    For the list of RGB color spaces supported, see the notes above;
+            type: enum COLOR_SPACE;\n
+            default: sRGB;
+        */
+    CV_WRAP void setColorSpace(COLOR_SPACE cs_);
```

Comment:
> `cs_`

No need to put underscores `_` in public headers (implementation can have own parameters names)

Reply:
Done